Source/Turbo Navigator/TurboNavigationHierarchyController.swift

```diff
@@ -168,21 +168,28 @@ class TurboNavigationHierarchyController {
         if navigationController.presentedViewController != nil {
             if modalNavigationController.viewControllers.count == 1 {
                 navigationController.dismiss(animated: proposal.animated)
-                delegate.refresh(navigationStack: .main)
+                refreshIfTopViewControllerIsVisitable(from: .main)
             } else {
                 modalNavigationController.popViewController(animated: proposal.animated)
-                delegate.refresh(navigationStack: .modal)
+                refreshIfTopViewControllerIsVisitable(from: .modal)
             }
         } else {
             navigationController.popViewController(animated: proposal.animated)
-            delegate.refresh(navigationStack: .main)
+            refreshIfTopViewControllerIsVisitable(from: .main)
+        }
+    }
+    
+    private func refreshIfTopViewControllerIsVisitable(from stack: NavigationStackType) {
+        if let navControllerTopmostVisitable = navController(for: stack).topViewController as? Visitable {
```

Comment:
We should discuss the best way to pass this refresh message to a non-Visitable view controller.

Reply:
As you mentioned on the call, leveraging the `PathConfigurationIdentifiable` protocol is an option here.